### PR TITLE
Bluetooth: controller: Implement a subset of Intel Mesh Vendor-Specific extensions

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

**Comment by carlescufi** [comment]
Initial implementation of the basic commands, with some limitations.

**Review by cvinayak — APPROVED** [review]
LGTM

**Review by cvinayak — CHANGES_REQUESTED** [review]
Need to do some testing

File: include/bluetooth/hci_vs.h [hunk 1]
```diff
@@ -163,6 +164,142 @@ struct bt_hci_evt_vs_scan_req_rx {
...
+struct bt_hci_mesh_pattern {
+	u8_t pattern_len;
+	u8_t pattern[0];
+} __packed;
+
+struct bt_hci_cp_mesh_set_scan_filter {
+	u8_t      scan_filter;
+	u8_t      filter_dup;
+	u8_t      num_patterns;
+	struct    bt_hci_mesh_pattern patterns[0];
```
Comment (cvinayak) [inline include/bluetooth/hci_vs.h]: @carlescufi This does not translate to Pattern_Length[i], Pattern[i] in the Spec document.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: include/bluetooth/hci_vs.h (hunk 1 above)
Comment (carlescufi) [reply include/bluetooth/hci_vs.h]: @cvinayak I will  fix this today

Comment (carlescufi) [reply include/bluetooth/hci_vs.h]: As checked with @holtmann, this is correct

File: subsys/bluetooth/controller/hci/hci.c [hunk 2]
```diff
@@ -1826,9 +1871,161 @@ static void vs_read_key_hierarchy_roots(struct net_buf *buf,
...
+	}
+
+	f = &scan_filters[filter];
+	for (i = 0; i < cmd->num_patterns; i++) {
+		if (!cmd->patterns[i].pattern_len ||
+		    cmd->patterns[i].pattern_len > BT_HCI_MESH_PATTERN_LEN_MAX) {
+			status = BT_HCI_ERR_INVALID_PARAM;
+			goto exit;
+		}
+		f->lengths[i] = cmd->patterns[i].pattern_len;
```
Comment (carlescufi) [inline subsys/bluetooth/controller/hci/hci.c]: this is wrong,  will fix

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by cvinayak** [comment]
I have only put in the new parameter and refactoring required, but random delay behavior is not implemented. Reason, it needs "thinking" to implement query of available window using the ticker and then applying a random delay inside the free window... it is like, in the delay range, a list of acceptable free windows needs to be identified, then a such random window needs to be picked, then a random offset inside that window.

**Comment by carlescufi** [comment]
Closing this because it will be re-submitted as part of the LL rearchitecture.

**Comment by summer20100514** [comment]
any updates here?

**Pull request closed without merging**